tlslite/utils/dns_utils.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2017 Hubert Kario
+#
+# See the LICENSE file for legal information regarding use of this file.
+
+import re
+
+"""Utilities for handling DNS hostnames"""
+
+
+def is_valid_hostname(hostname):
+    """
+    Check if the parameter is a valid hostname.
+
+    @type hostname: str or bytearray
+    @rtype: boolean
+    """
+    try:
+        if not isinstance(hostname, str):
+            hostname = hostname.decode('ascii', 'strict')
+    except UnicodeDecodeError:
+        return False
+    if hostname[-1] == ".":
+        # strip exactly one dot from the right, if present
+        hostname = hostname[:-1]
+    if len(hostname) > 253:
```

Comment:
comment added. Since the value depends on the final dot being stripped, I've decided against creating a constant